crowbar_framework/app/models/service_object.rb

```diff
@@ -1178,7 +1181,7 @@ def apply_role(role, inst, in_queue)
       nodes_in_batch = []
 
       elems.each do |role_name|
-        old_nodes = old_elements[role_name]
+        old_nodes = old_elements[role_name] || databag["deployment"][@bc_name]["elements"]["to_remove"].keys
```

Comment:
@vuntz this would give us any old_nodes that were left over after a crash, but just if apply role runs in the context of this barclamp... do you think that is enough? or should we gather the leftover nodes from **all** barclamps?